Adds support for the SH4 to the embedded QT build. This patch 
doesn't add support for the atomic operations as yet, it uses the
generic posix mutex implementation. From experiments done on the ST200
implementing this gives about 10% performance increase.

Signed-off-by: David McKay <david.mckay@st.com>

Index: qt-everywhere-opensource-src-4.6.0/mkspecs/qws/linux-sh4-g++/qmake.conf
===================================================================
--- /dev/null
+++ qt-everywhere-opensource-src-4.6.0/mkspecs/qws/linux-sh4-g++/qmake.conf
@@ -0,0 +1,29 @@
+#
+# qmake configuration for linux-g++ using the sh4-linux-g++ crosscompiler
+#
+
+include(../../common/g++.conf)
+include(../../common/linux.conf)
+include(../../common/qws.conf)
+
+QMAKE = $$[QT_INSTALL_BINS]/sh4-linux-qmake
+
+# modifications to g++.conf
+QMAKE_CC                = sh4-linux-gcc
+QMAKE_CXX               = sh4-linux-g++
+QMAKE_LINK              = sh4-linux-g++
+QMAKE_LINK_SHLIB        = sh4-linux-g++
+QMAKE_CFLAGS_RELEASE    = -Os
+QMAKE_CXXFLAGS_RELEASE	 = -Os
+
+# modifications to linux.conf
+QMAKE_AR                = sh4-linux-ar cqs
+QMAKE_OBJCOPY           = sh4-linux-objcopy
+QMAKE_RANLIB            = sh4-linux-ranlib
+QMAKE_STRIP             = true
+QMAKE_MOC               = $$[QT_INSTALL_BINS]/sh4-linux-moc
+QMAKE_UIC               = $$[QT_INSTALL_BINS]/sh4-linux-uic
+
+QMAKE_RCC               = $$[QT_INSTALL_BINS]/sh4-linux-rcc
+
+load(qt_config)
Index: qt-everywhere-opensource-src-4.6.0/mkspecs/qws/linux-sh4-g++/qplatformdefs.h
===================================================================
--- /dev/null
+++ qt-everywhere-opensource-src-4.6.0/mkspecs/qws/linux-sh4-g++/qplatformdefs.h
@@ -0,0 +1,38 @@
+/****************************************************************************
+**
+** Copyright (C) 2008 Nokia Corporation and/or its subsidiary(-ies).
+** Contact: Qt Software Information (qt-info@nokia.com)
+**
+** This file is part of the qmake spec of the Qt Toolkit.
+**
+** Commercial Usage
+** Licensees holding valid Qt Commercial licenses may use this file in
+** accordance with the Qt Commercial License Agreement provided with the
+** Software or, alternatively, in accordance with the terms contained in
+** a written agreement between you and Nokia.
+**
+**
+** GNU General Public License Usage
+** Alternatively, this file may be used under the terms of the GNU
+** General Public License versions 2.0 or 3.0 as published by the Free
+** Software Foundation and appearing in the file LICENSE.GPL included in
+** the packaging of this file.  Please review the following information
+** to ensure GNU General Public Licensing requirements will be met:
+** http://www.fsf.org/licensing/licenses/info/GPLv2.html and
+** http://www.gnu.org/copyleft/gpl.html.  In addition, as a special
+** exception, Nokia gives you certain additional rights. These rights
+** are described in the Nokia Qt GPL Exception version 1.3, included in
+** the file GPL_EXCEPTION.txt in this package.
+**
+** Qt for Windows(R) Licensees
+** As a special exception, Nokia, as the sole copyright holder for Qt
+** Designer, grants users of the Qt/Eclipse Integration plug-in the
+** right for the Qt/Eclipse Integration to link to functionality
+** provided by Qt Designer and its related libraries.
+**
+** If you are unsure which license is appropriate for your use, please
+** contact the sales department at qt-sales@nokia.com.
+**
+****************************************************************************/
+
+#include "../../linux-g++/qplatformdefs.h"
Index: qt-everywhere-opensource-src-4.6.0/src/3rdparty/webkit/JavaScriptCore/wtf/Platform.h
===================================================================
--- qt-everywhere-opensource-src-4.6.0.orig/src/3rdparty/webkit/JavaScriptCore/wtf/Platform.h
+++ qt-everywhere-opensource-src-4.6.0/src/3rdparty/webkit/JavaScriptCore/wtf/Platform.h
@@ -378,6 +378,14 @@
 #   include <ce_time.h>
 #endif
 
+/* PLATFORM(SH4) */
+#if defined(__sh__)
+#define WTF_PLATFORM_SH 1
+#if defined(__SH4__)
+#define WTF_PLATFORM_SH4 1
+#endif
+#endif
+
 /* PLATFORM(ST200) */
 #if defined(__st200__)
 #define WTF_PLATFORM_ST200 1
